pkg/api/rest/create.go

```diff
@@ -57,7 +57,14 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx api.Context, obj runtime.Obje
 			return errors.NewBadRequest("the namespace of the provided object does not match the namespace sent on the request")
 		}
 	} else {
-		objectMeta.Namespace = api.NamespaceNone
+		if ns, ok := api.NamespaceFrom(ctx); ok && ns != api.NamespaceNone {
+			return errors.NewBadRequest("namespace is not allowed on this type")
```

Comment:
I think I would change the message to say:

"When creating an object with kind %v, you must have a context with no namespace."

Basically, not sure if type is better than kind, and I imagine this error is not something an average end-user of system will ever find, and is more likely something someone writing a bad unit test would encounter.